src/uwtools/config.py

```diff
@@ -0,0 +1,198 @@
+'''
+This file contains the Config file and its subclasses for a variety of
+dicatable file types.
+'''
+
+import abc
+import collections
+import configparser
+
+import f90nml
+import yaml
+
+class Config(collections.UserDict):
+
```

Comment:
This is an abstract base class as collections objects are derived from ABCs ([source](https://docs.python.org/3/library/abc.html)) 

> The [collections](https://docs.python.org/3/library/collections.html#module-collections) module has some concrete classes that derive from ABCs; these can, of course, be further derived. In addition, the [collections.abc](https://docs.python.org/3/library/collections.abc.html#module-collections.abc) submodule has some ABCs that can be used to test whether a class or instance provides a particular interface, for example, if it is hashable or if it is a mapping.


Just to double check the documentation wasn't sending me down a questionable path, I also followed the source code for collections through the definition of collections.UserDict [here](https://github.com/python/cpython/blob/230b630a767a457c6e7e4d797548a172c64ae735/Lib/collections/__init__.py) and it's base class defined in collections' Lib [here](https://github.com/python/cpython/blob/230b630a767a457c6e7e4d797548a172c64ae735/Lib/_collections_abc.py#L1).

Given that we are inheriting a wrapped ABC, I don't think we'll have problems.

Reply:
Yes you are close,  it is true  UserDict ⊂ collections and ABC ⊂ collections but by no means is collections.ABC ⊂ collections.UserDict and you don't get all the "base classes defined in collections" when inheriting collections.UserDict.  Having said that this is what should be done: Config(ABC)  -> childClass(Config, UserDict) which I had tested before I had this better understanding [[ref](https://stackoverflow.com/questions/66173332/python-abc-multiple-inheritance)]

Reply:
This seems like a question with a concrete answer - is there a test that would/does confirm if this is behaving as expected?

Reply:
Oh and incidentally Config had been inheriting UserDict only because we where overloading the **__getattr__** method in order to support the dot as a value resolution operator and it appears you do not have that in this PR.

Reply:
@TerrenceMcGuinness-NOAA the links I included don't imply the relationships you suggest here at all. They define that the `collections.UserDict` has ABCMeta as one of its base classes, therefore `Config` inherits ABCMeta as a base class.

@benjamin-cash There is indeed. Here I show the example as in the one in the StackOverflow Terry linked above:
```
from abc import ABC, abstractmethod
class BaseABC(ABC):
    @abstractmethod
    def hello(self):
        pass
class Child(BaseABC):
    pass

child = Child()
```

Just like it says in SO, this code will fail if the `Child` class does not implement the `hello` method. I tested it out by removing ABC as the base class and I found that the code will not complain at all about `Child` not implementing `hello`. We can test that with a version of YAMLConfig in a similar way. In my local code, I have commented out the YAMLConfig method `_load`, rebuilt with pip, and tried to instantiate a YAMLConfig object. Here is the code and the stack trace:

```
>>> import uwtools.config as config
>>> cfg = config.YAMLConfig()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: Can't instantiate abstract class YAMLConfig with abstract methods _load
```

Since `collections.UserDict` is derived from a `ABCMeta` as you can clearly follow in the source code, using the `@abstractmethod` decorator works just as expected.

To addres your follow up, Terry, `collections.UserDict` is not used "only because we where overloading the **getattr** method". That is a convenient feature that we _can_ use and I have chosen not to implement here because as Rahul pointed out, it was already a big push. It can always be added with a smaller PR in the future. The reason that we are using `collections.UserDict` as a base class is because we'd like to have some additional methods to manipulate these config dictionaries and this is the Pythonic way to achieve that goal -- extend the functionality of the built-in dictionary.

Reply:
Ok, given that there is a concrete test showing that the code is behaving as expected, and this code was already approved once, I feel that this PR should be approved and any additional concerns can be addressed in future tickets and PRs.

Reply:
@TerrenceMcGuinness-NOAA Are you satisfied with the test @christinaholtNOAA provided here?

Reply:
Yes will like to see dot resolution at all costs myself because from users perspective it had often been called out as a "favorite".  Still the "proper" way to establish a Abstract Base Class is to inherit the ABCMeta class directly and not to simply subsume it from a "derived" class.    -- end of suggestion --

Reply:
Yes Benjamin, I like you "common sense" approach.  Yes we all now have a good understanding of the issues and intent, as well as why I still have a relevant prototype branch, so let's move on shall we :)